### DISCUSSION TITLE
Project's xpacks folder hierarchy

### DISCUSSION THREAD

**Discussion by jsineau-asulab** [xpack/xpm-js#210] [Q&A] [answered]
Hi,

The project's _/xpacks_ folder hierarchy isn't similar if the project has xPacks from only one "category" (for instance, [3rd party source xPacks](https://github.com/xpack-3rd-party/)) or has xPacks from several "categories" (for instance, [3rd party source xPacks](https://github.com/xpack-3rd-party/) and [xPack Binary Development Tools](https://github.com/xpack-dev-tools/)).

This is pitty if we use the relative path of a xPack file, for instance to use an extension of the [Doxygen awesome css](https://github.com/xpack-3rd-party/doxygen-awesome-css-xpack).

The [user guide](https://github.com/jothepro/doxygen-awesome-css/blob/main/docs/extensions.md#installation) of this project indicates:

> Add the required resources in your Doxyfile:
> HTML_EXTRA_FILES: doxygen-awesome-darkmode-toggle.js

The _doxygen-awesome-darkmode-toggle.js_ is in the _xpack-3rd-party-doxygen-awesome-css_ folder.

Have you any idea to solve it? May be an option of `xpm` that I don't know about.

In advance, thank you.

### Hierarchies examples

Hierarchy with only one "category":
```
my-project
  |_ xpacks
    |_ xpack-3rd-party-doxygen-awesome-css
```

Hierarchy with several "categories":
```
my-project
  |_ xpacks
    |_ @xpack-3rd-party
      |_ xpack-3rd-party-doxygen-awesome-css
    |_ @xpack-dev-tools
      |_ meson-build
```

**Comment by ilg-ul** [comment, marked as answer]
Hi @jsineau-asulab,

I'm not sure I understand your questions.

Are you using the latest xpm and your project is configured with a recent `minimumXpmRequired` in your `package.json`?

The strategy used for creating sub-folders changed at a certain point, and now it should look like:

```
my-project
  |_ xpacks
    |_ @xpack-3rd-party
      |_ doxygen-awesome-css
    |_ @xpack-dev-tools
      |_ meson-build
```

in other words, the npm `@scope` is preserved and the name is not linearised.

**Comment by ilg-ul** [comment]
As an example of using the `doxygen-awesome-css` in a project, you can take a look at:

- https://github.com/micro-os-plus/utils-lists-xpack

**Comment by jsineau-asulab** [comment]
Hi @ilg-ul,

I understand now. I started my project with a copy/paste of the [package.json](https://github.com/xpack-3rd-party/doxygen-awesome-css-xpack/blob/xpack/package.json) from the [doxygen-awesome-css-xpack](https://github.com/xpack-3rd-party/doxygen-awesome-css-xpack).

Within this  [package.json](https://github.com/xpack-3rd-party/doxygen-awesome-css-xpack/blob/xpack/package.json) the `minimumXpmRequired` parameter isn't up to date, `0.15.0` and trig my previous behavior.

Therefore, I shall update this parameter from `0.15.0` to `0.18.0` and the project's xpacks folder hierarchy is right.

Do you want that I update your project?

🙏 Thank you Liviu and my apologies for my english

**Reply by ilg-ul** [reply]
> Do you want that I update your project?

I'll update it at a certain moment.

**Reply by ilg-ul** [reply]
I made a new release of the @xpack-3rd-party/doxygen-awesome-css package.
